nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/AbstractSNMPProcessor.java

```diff
@@ -252,6 +252,7 @@ public void close() {
             this.getLogger().warn("Failure while closing UDP transport mapping", e);
         }
         this.snmp = null;
+        this.snmpTarget = null;
```

Comment:
along with this change, I was wondering move below code block from ontrigger to onscheduled?
        
 synchronized (this) {
             this.buildTargetResource(context);
         }